SpellChecker.java

```diff
@@ -8,28 +8,70 @@ public static void main(String[] args) {
 		String[] dictionary = readDictionary("dictionary.txt");
 		String correction = spellChecker(word, threshold, dictionary);
 		System.out.println(correction);
+
 	}
 
 	public static String tail(String str) {
-		// Your code goes here
+		String tail = "";
+		if (str.length() == 1){
+			 return tail;
+		}
+		tail = str.substring(1);
+		return tail;
 	}
 
 	public static int levenshtein(String word1, String word2) {
-		// Your code goes here
+		word1 = word1.toLowerCase();
+		word2 = word2.toLowerCase();
+		int dist = 0;
+		if (word1.length() == 0){
+			dist = word2.length();
+		}
+		else if (word2.length() == 0){
+			dist = word1.length();
+		}
+		else if (word1.charAt(0) == word2.charAt(0)){
+			dist = levenshtein(tail(word1), tail(word2));
+		}
+		
+		else{	
+			int a = levenshtein(tail(word1), word2);
+			int b = levenshtein(word1, tail(word2));
+			int c = levenshtein(tail(word1), tail(word2));
+			int min = Math.min(a, b);
+			if (c < min){
+				 min = c;
+			}
```

Comment:
Lets be consistent with the way calculating min :)
```suggestion
			int min = Math.min(a, b);
			min = Math.min(min, c);
```